packages/headless/src/controllers/commerce/context/headless-context.ts

```diff
@@ -0,0 +1,212 @@
+import { Schema } from "@coveo/bueno";
+import {requiredNonEmptyString} from '../../../utils/validate-payload';
+import {CommerceEngine} from '../../../app/commerce-engine/commerce-engine';
+import {buildController, Controller} from '../../controller/headless-controller';
+
+const optionsSchema = new Schema({
+  url: requiredNonEmptyString,
+});
+
+export interface User {
+  userId?: string;
+  email?: string;
+  userIp?: string;
+  userAgent?: string;
+}
+
+export interface View {
+  url: string;
+}
+
+export interface Product {
+  groupId?: string;
+  productId?: string;
+  sku?: string;
+}
+
+export interface CartItem {
+  product: Product;
+  quantity: number;
+}
+
+export interface ContextOptions {
+  trackingId?: string;
+  language?: string;
+  currency?: string;
+  // TODO: Does it make sense for the clientId to be part of the context state?
+  clientId?: string;
+  user?: User;
+  view?: Partial<View>;
```

Comment:
I think `view` should be set on the ProductListing controller, not on the "global" commerce context

Reply:
Same thing for `labels` (although I believe we decided not to expose this in Headless initially)

Reply:
I removed `labels` but left `view` for now since we're still figuring out which controller should handle the `url`.